std/asyncio/system/Signal.hx

```diff
@@ -0,0 +1,47 @@
+package asyncio.system;
+
+/**
+	Signals for inter-process communication.
+	Signals are sent by operating system or by a process to another process.
+	Also, a process can signal itself.
+
+	TODO: any other signals to have own constructors here?
+**/
+enum Signal {
+	/**
+		Terminate a process.
+		The process is _not_ immediately killed.
+		Instead the process can handle this signal to gracefully shutdown
+		(remove temporary files, close socket connections etc).
+		POSIX equivalent: SIGTERM
+	**/
+	Terminate;
+	/**
+		Immediately terminate a process.
+		The process cannot handle this signal.
+		That means, for example, temporary files may stay in file system.
+		POSIX equivalent: SIGKILL
+	**/
+	Kill;
+	/**
+		Interrupt a process.
+		The same as pressing "CTRL+C" in a terminal.
+		POSIX equivalent: SIGINT
+	**/
+	Interrupt;
+	/**
+		_Pause_ a process.
+		The process cannot handle this signal.
+		POSIX equivalent: SIGSTOP
+	**/
+	Stop;
+	/**
+		Continue previously stopped process.
+		POSIX equivalent: SIGCONT
+	**/
+	Resume;
+	/**
+		Any signal can be specified by its code.
+	**/
+	Custom(signal:Int);
```

Comment:
Perhaps it makes sense to have an "open" `enum abstract` over `Int` then?

Reply:
What do you mean? `from Int to Int`?

Reply:
What do you mean? `from Int to Int`?

Reply:
Indeed. I mean, it's no biggie, but seeing how it can be any int, with some of them having well-defined meanings, it seems a more natural fit. Otherwise there will have to be mapping from/to `Int` and every implementor will have to put it somewhere.

Reply:
The issue is signal codes are not cross-platform. And some signals even may have multiple codes on the same system.

Reply:
Added `CSigNum` enum abstract.

Reply:
Where does that obscure-seeming name come from? Why not simply `extern enum abstract Signal(Int)` or `SignalCode`?

Reply:
From `man signal`. I'm ok with `CSignalCode`. I wan't to keep `C` prefix as a note that those are taken directly from C API.

Reply:
Meh... so should everything in the Python package be prefixed with `P`, and everything in the Flash package with `F`, etc? :D

Reply:
I do think this situation is different from Python and Flash because it is something that will be the same in multiple platforms. Having said that, even though it comes from a C library, a better prefix might be `Posix`.